lib/data/external/GcpClient.js

```diff
@@ -91,6 +102,201 @@ class GcpClient extends AwsClient {
             return callback(null, gcpResp);
         });
     }
+
+    createMPU(key, metaHeaders, bucketName, websiteRedirectHeader, contentType,
+        cacheControl, contentDisposition, contentEncoding, log, callback) {
+        const metaHeadersTrimmed = {};
+        Object.keys(metaHeaders).forEach(header => {
+            if (header.startsWith('x-amz-meta-')) {
+                const headerKey = header.substring(11);
+                metaHeadersTrimmed[headerKey] = metaHeaders[header];
+            }
+        });
+        Object.assign(metaHeaders, metaHeadersTrimmed);
+        const gcpKey = this._createGcpKey(bucketName, key, this._bucketMatch);
+        const params = {
+            Bucket: this._mpuBucketName,
+            Key: gcpKey,
+            Metadata: metaHeaders,
+            ContentType: contentType,
+            CacheControl: cacheControl,
+            ContentDisposition: contentDisposition,
+            ContentEncoding: contentEncoding,
+        };
+        return this._client.createMultipartUpload(params, (err, mpuResObj) => {
+            if (err) {
+                logHelper(log, 'error', 'err from data backend',
+                  err, this._dataStoreName, this.clientType);
+                return callback(errors.ServiceUnavailable
+                  .customizeDescription('Error returned from ' +
+                  `GCP: ${err.message}`)
+                );
+            }
+            return callback(null, mpuResObj);
+        });
+    }
+
+    completeMPU(jsonList, mdInfo, key, uploadId, bucketName, log, callback) {
+        const gcpKey = this._createGcpKey(bucketName, key, this._bucketMatch);
+        const partArray = [];
+        const partList = jsonList.Part;
+        for (let i = 0; i < partList.length; ++i) {
+            const partObj = partList[i];
+            if (!partObj.PartNumber || !partObj.ETag) {
+                return callback(errors.MalformedXML);
+            }
+            const number = partObj.PartNumber[0];
+            const partNumber = typeof number === 'string' ?
```

Comment:
minor: you can also use `parseInt` on integers, it just returns it as is, so you could remove the condition.